hydro/exchange/containers.py

```diff
@@ -721,3 +724,93 @@ def to_xarray(self):
+            format_dict[param.whp_name] = (param.field_width, param.numeric_precision)
+            na_values[param.whp_name] = f"{-999:>{param.field_width}}"
+            if param in self.flags:
+                units[f"{param.whp_name}_FLAG_W"] = ""
+                format_dict[f"{param.whp_name}_FLAG_W"] = (1, 0)  # flags are integers
+                na_values[f"{param.whp_name}_FLAG_W"] = "9"  # flag field_width is 1
+
+        header = f"""{file_indicator}
+{",".join(str(k) for k in units.keys())}
+{",".join(str(v) for v in units.values())}
+"""
+
+        # consolidate to dataframe
+        df_list = []
+        for p in self.iter_profiles():
+            df = pd.DataFrame()
+            for param in self.parameters:
+                if param.whp_name in ["DATE", "TIME"]:
+                    df[param.whp_name] = p.time_to_ndarray().flatten()
+                else:
+                    df[param.whp_name] = p.parameter_to_ndarray(param).flatten()
+                if param in self.flags:
+                    df[param.whp_name + "_FLAG_W"] = p.flag_to_ndarray(param).flatten()
+            df_list.append(df)
+
+        if self.file_type == FileType.BOTTLE:
+            df_list = [pd.concat(df_list)]
+
+        for df in df_list:
+            # fix formatting
+            df["DATE"] = df["DATE"].dt.strftime("%Y%m%d")
+            df["TIME"] = df["TIME"].dt.strftime("%H%M")
+            for key, (width, prec) in format_dict.items():
+                if key in ["DATE", "TIME"]:
+                    continue
+                elif df[key].dtype == object:
+                    df[key] = df[key].map(f"{{:>{width}s}}".format, na_action="ignore")
+                    continue
+                df[key] = df[key].map(  # if prec is None: prec = 0
+                    f"{{:>{width}.{prec or 0}f}}".format, na_action="ignore"
+                )
+            df.fillna(value=na_values, inplace=True)
+            data_bytes_csv = bytes(df.to_csv(index=False, header=False).encode("utf8"))
+            file_contents = header.encode("utf8") + data_bytes_csv + b"END_DATA"
+
+            # save
+            expocode = df["EXPOCODE"].unique().item().strip()
+            postfix = {"BOTTLE": "_hy1", "CTD": "_ct1"}[self.file_type.name]
+            folder = Path()
+            if self.file_type == FileType.CTD:
+                station = int(df["STNNBR"].unique().item())
+                cast = int(df["CASTNO"].unique().item())
+                infix = f"_{station:05d}_{cast:05d}"
+                fname = Path(expocode + infix + postfix + ".csv")
+                if isinstance(filename_or_obj, (str, Path)):
+                    folder = Path(filename_or_obj).with_suffix("")
+                elif not filename_or_obj:
+                    folder = Path(expocode + postfix)
+                if not zip_ctd:
+                    folder.mkdir(exist_ok=True)
+                else:
+                    with ZipFile(folder.with_suffix(".zip"), "a") as zipped:
```

Comment:
The default is no compression, not sure what the best method is

Reply:
"standard" zip compression method is DEFLATED